examples/d2.jl

```diff
@@ -3,8 +3,10 @@ using Chmy, KernelAbstractions
 @kernel function d2!(V1, C2, C4, C1, g::StructuredGrid, O)
     I = @index(Global, NTuple)
     I = I + O
+    Il = I .- (1,)
+    Ir = I .+ (1,)
     V1[I...] = ∂x(C1, g, I...)
-    C2[I...] = ((right(C1, (Center(),), (Center(),), Dim(1), I...) - C1[I...]) * iΔ(g, Center(), Dim(1), I...) - ((C1[I...] - left(C1, (Center(),), (Center(),), Dim(1), I...)) * iΔ(g, Center(), Dim(1), I...))) * iΔ(g, Center(), Dim(1), I...)
+    C2[I...] = ((right(C1, (Center(),), (Center(),), Dim(1), I...) - C1[I...]) * iΔ(g, Vertex(), Dim(1), Ir...) - ((C1[I...] - left(C1, (Center(),), (Center(),), Dim(1), Il...)) * iΔ(g, Vertex(), Dim(1), I...))) * iΔ(g, Center(), Dim(1), I...)
```

Comment:
you could get rid of one memread by loading `C1ᵢ = C1[I...]`

Reply:
Sure - thanks for the suggestion! This line though is rather to test "correctness" as to see whether the API I am trying to add provides the expected results.

Currently, it does, but this is mostly because we are using a `UniformGrid` with scalar or constant spacing etc... I am wondering now if there is an elegant way to generalise the second derivative at least for StructuredGrids, which would require non constant grid spacing and thus it would matter to take the correct `∆x` during the process.

Maybe @utkinis could provide some direction on whether the API allows for concise way of selecting `I∆` for shifted `I...`. The `loc` part one could achieve using the `flipped` function.